test/parallel/test-console-table.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+require('../common');
+
+const assert = require('assert');
+const { Console } = require('console');
+
+const queue = [];
+
+const console = new Console({ write: (x) => {
+  queue.push(x);
+}, removeListener: () => {} }, process.stderr, false);
+
+console.table({ a: 1 });
+assert.strictEqual(queue.shift(), '(index)	Values\na	1\n');
+
+console.table([{ a: 1 }, { b: 2 }])
+assert.strictEqual(queue.shift(), `(index)	a	b
+0	1	undefined
+1	undefined	2
+`);
```

Comment:
Needs more tests:

```js
console.table([3, 4, { a: 3 }]);
console.table([3, 4, { a: 3 }], ['nonexistent']);
console.table([3, 4, { a: 3 }], ['Value']);
console.table([3, 4, { Value: 3 }], ['Value']);
console.table([,,,,,,,,,4]);
console.table([[,,,,,,,,,4]]);
console.table([[,,,,,,,,,{}]]);

// etc.

// Check browsers for sensible behaviors.
```

Also needed to test are Map objects, Map Iterator objects, Set objects, Set Iterator objects.